### ISSUE TITLE
Slack integration breaks on alert from vulnerability detector

### ISSUE THREAD

**Issue by harkylton** [wazuh/wazuh#736]
The slack integration script breaks on alerts from vulnerability detector since these alerts don't have `full_log`. The problem is not logged anywhere unless you set `debug_enabled = True` in the script. Since `str` is applied to the exception, all you get in the log is the arguments of the Exception i.e. `'full_log'`. Logging the traceback would be helpful. When running the script manually, the exception is printed correctly since it's re-raised. After occurring once, the integration seems to also stop working for all other alerts, even those that do have `full_log`.

## Example alert
```json
{"manager": {"name": "wazuh-manager"}, "rule": {"firedtimes": 1, "description": "CVE-2018-11235 on Ubuntu 16.04 LTS (xenial) - high.", "level": 10, "groups": ["vulnerability-detector"], "mail": false, "id": "23505"}, "decoder": {"name": "json"}, "timestamp": "2018-06-07T08:09:23+0000", "data": {"vulnerability": {"updated": "2018-05-30", "cve": "CVE-2018-11235", "severity": "High", "reference": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-11235", "title": "CVE-2018-11235 on Ubuntu 16.04 LTS (xenial) - high.", "package": {"version": "1:2.7.4-0ubuntu1.3", "name": "git", "condition": "less than 1:2.7.4-0ubuntu1.4"}, "state": "Fixed", "published": "2018-05-30"}}, "id": "1528358963.382493"}, "id": "001", "name": "server1"}, "location": "vulnerability-detector"}
```

## How to reproduce
```bash
python /var/ossec/integrations/slack "/tmp/alert.json" "" "<webhook>"
```

## Output
```
Thu Jun 07 08:25:27 UTC 2018: # Starting

Thu Jun 07 08:25:27 UTC 2018: # Webhook
Thu Jun 07 08:25:27 UTC 2018: <webhook>

Thu Jun 07 08:25:27 UTC 2018: # File location
Thu Jun 07 08:25:27 UTC 2018: /tmp/alert.json

Thu Jun 07 08:25:27 UTC 2018: # Processing alert
Thu Jun 07 08:25:27 UTC 2018: {u'timestamp': u'2018-06-07T08:09:23+0000', u'rule': {u'firedtimes': 1, u'description': u'CVE-2018-11235 on Ubuntu 16.04 LTS (xenial) - high.', u'level': 10, u'groups': [u'vulnerability-detector'], u'mail': False, u'id': u'23505'}, u'manager': {u'name': u'wazuh-manager'}, u'id': u'001', u'name': u'server1'}, u'decoder': {u'name': u'json'}, u'data': {u'vulnerability': {u'updated': u'2018-05-30', u'severity': u'High', u'reference': u'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-11235', u'package': {u'version': u'1:2.7.4-0ubuntu1.3', u'name': u'git', u'condition': u'less than 1:2.7.4-0ubuntu1.4'}, u'title': u'CVE-2018-11235 on Ubuntu 16.04 LTS (xenial) - high.', u'state': u'Fixed', u'published': u'2018-05-30', u'cve': u'CVE-2018-11235'}}, u'id': u'1528358963.382493', u'location': u'vulnerability-detector'}

Thu Jun 07 08:25:27 UTC 2018: # Generating message
Thu Jun 07 08:25:27 UTC 2018: 'full_log'

Traceback (most recent call last):
  File "/var/ossec/integrations/slack", line 128, in <module>
    main(sys.argv)
  File "/var/ossec/integrations/slack", line 59, in main
    msg = generate_msg(json_alert)
  File "/var/ossec/integrations/slack", line 89, in generate_msg
    msg['text'] = alert['full_log']
KeyError: 'full_log'
```

## Temporary fix
A temporary fix is to allow `full_log` to be `None`, but it would be nicer if the vulnerability detector actually had a `full_log` explaining the content of `data.vulnerability` in a readable way. 

```diff
@@ -86,7 +86,7 @@ def generate_msg(alert):
     msg['color'] = color
     msg['pretext'] = "WAZUH Alert"
     msg['title'] = alert['rule']['description']
-    msg['text'] = alert['full_log']
+    msg['text'] = alert.get('full_log')
     agent = { "title":"Agent", "value":"({0}) - {1}".format(alert['agent']['id'],alert['agent']['name']) }
     location = { "title":"Location", "value":alert['location'] }
     rule = { "title":"Rule ID", "value":"{0} _(Level {1})_".format(alert['rule']['id'],level) }
```

**Comment by harkylton** [comment]
Any insights here @vikman90? How should I proceed? I can make a PR if that's desirable. I have to manually mitigate this whenever we upgrade the wazuh manager.

**Comment by vikman90** [comment]
Hi @harkylton,

This is a bug, indeed. Alerts from Vulnerability Detector don't have full log because it's disabled in the rules: https://github.com/wazuh/wazuh/blob/master/etc/rules/0520-vulnerability-detector.xml#L10. This is because the `full_log` is the raw input of the event, in this case, a JSON string, that would duplicate the information in the alert. This is producing the bug in the Slack integration.

However, we are working on a new iteration of the Vulnerability Detector that will add more descriptive data to the alert, but it's not the full log, it will come in another field.

I will apply your fix in a PR and propose it for the next version 3.4.

Thank you very much for reporting this issue.
Best regards.

**Comment by vikman90** [comment]
Hi @harkylton,

We have already applied your fix in the branch 3.4 for the next release, v3.4.0. We have tested it and it works.

Let me close this issue. Thank you very much for your report and your fix!

Best regards.
